umh-core/pkg/service/topicbrowser/parse_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topicbrowser
+
+import (
+	"bytes"
+	"encoding/hex"
+	"strconv"
+	"time"
+
+	"github.com/pierrec/lz4/v4"
+	s6svc "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/s6"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("extractRaw / parseBlock", func() {
+	var (
+		payload    []byte      // original message
+		compressed []byte      // LZ4-compressed block
+		epochMS    int64       // timestamp used in the test logs
+		rb         *Ringbuffer // tiny buffer for the service
+		service    *Service    // minimal service with just the ring-buffer
+	)
+
+	// helper to build a full, well-formed log slice
+	buildLogs := func(includeTimestamp bool, dataLine string) []s6svc.LogEntry {
```

Comment:
maybe make this not an inline fucntion, but a proper function, and write down for what this is used. 

because I assume this is also used not only for well formed log slices, but also to produce bad data using includetimestamp=false